tests/integration/servicecatalog/cmd_service_test.go

```diff
@@ -435,4 +435,27 @@ var _ = Describe("odo service command tests", func() {
 			helper.CmdShouldPass("odo", "catalog", "describe", "service", "dh-import-vm-apb")
 		})
 	})
+
+	Context("When the application is deleted", func() {
+		JustBeforeEach(func() {
+			preSetup()
+			app = helper.RandString(6)
+		})
+		It("should delete the service(s) in the application as well", func() {
+			helper.CmdShouldPass("odo", "service", "create", "--app", app, "-w", "dh-postgresql-apb", "--project", project, "--plan", "dev",
+				"-p", "postgresql_user=luke", "-p", "postgresql_password=secret",
+				"-p", "postgresql_database=my_data", "-p", "postgresql_version=9.6")
+			ocArgs := []string{"get", "serviceinstance", "-o", "name", "-n", project}
+			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+				return strings.Contains(output, "dh-postgresql-apb")
+			})
+
+			helper.CmdShouldPass("odo", "app", "delete", app, "-f")
+
+			ocArgs = []string{"get", "serviceinstances"}
+			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
+				return strings.Contains(output, "Deprovisioning")
```

Comment:
```Deprovisioning``` literally seems the process is still going on. Mab be ```Deprovisioned``` would be the best fit for this. But it seems we don't have control over the ```oc``` output. So @dharmit Can you please confirm does the substring ```Deprovisioning``` means all the services link to the application are deleted.

Reply:
`Deprovisioning` indeed means that the process of deletion of service is still going on. Once the service is deleted, the output of `oc get serviceinstances` becomes something like `No resources found...` since there are no service instances in the project. It won't print `Deprovisioned` for deleted service.

Reply:
I'm trying to see if doing `ocArgs = []string{"get", "serviceinstances", "2>&1"}` helps. So far, it doesn't seem to be working on my test setup.

Reply:
Well, trying to redirect stderr to stdout is not working:
```sh
Running oc with args [oc get serviceinstances 2>&1]
[oc] Error from server (NotFound): serviceinstances.servicecatalog.k8s.io "2>&1" not found

------------------------------

Failure [20.698 seconds]
odo service command tests
/home/dshah/go/src/github.com/openshift/odo/tests/integration/servicecatalog/cmd_service_test.go:15
  When the application is deleted
  /home/dshah/go/src/github.com/openshift/odo/tests/integration/servicecatalog/cmd_service_test.go:439
    should delete the service(s) in the application as well [It]
    /home/dshah/go/src/github.com/openshift/odo/tests/integration/servicecatalog/cmd_service_test.go:444

    No future change is possible.  Bailing out early after 0.111s.
    Running oc with args [oc get serviceinstances 2>&1]
    Expected
        <int>: 1
    to match exit code:
        <int>: 0

    /home/dshah/go/src/github.com/openshift/odo/tests/helper/helper_run.go:34
```
Doesn't work when I use `odo` command either:
```sh
[odo]  ✗  unknown command "2>&1" for "odo service list"

------------------------------
• Failure [19.730 seconds]
odo service command tests
/home/dshah/go/src/github.com/openshift/odo/tests/integration/servicecatalog/cmd_service_test.go:15
  When the application is deleted
  /home/dshah/go/src/github.com/openshift/odo/tests/integration/servicecatalog/cmd_service_test.go:439
    should delete the service(s) in the application as well [It]
    /home/dshah/go/src/github.com/openshift/odo/tests/integration/servicecatalog/cmd_service_test.go:444

    No future change is possible.  Bailing out early after 0.040s.
    Running odo with args [odo service list --app kushnf 2>&1]
    Expected
        <int>: 1
    to match exit code:
        <int>: 0

    /home/dshah/go/src/github.com/openshift/odo/tests/helper/helper_run.go:34
```

Looks like using shell way of redirecting stderr to stdout doesn't work well from Go code. :disappointed: